test/unit/org/apache/cassandra/locator/ReplicaCollectionTest.java

```diff
@@ -141,28 +147,37 @@ public void testOrderOfIteration()
         {
             Assert.assertEquals(canonicalList, ImmutableList.copyOf(test));
             Assert.assertEquals(canonicalList, test.stream().collect(Collectors.toList()));
-            Assert.assertEquals(new LinkedHashSet<>(Lists.transform(canonicalList, Replica::endpoint)), test.endpoints());
+            Assert.assertTrue(Iterables.elementsEqual(new LinkedHashSet<>(Lists.transform(canonicalList, Replica::endpoint)), test.endpoints()));
         }
 
         private void assertSubList(C subCollection, int from, int to)
         {
-            Assert.assertTrue(subCollection.isSnapshot);
             if (from == to)
             {
                 Assert.assertTrue(subCollection.isEmpty());
             }
             else
             {
-                List<Replica> subList = this.test.list.subList(from, to);
-                if (test.isSnapshot)
-                    Assert.assertSame(subList.getClass(), subCollection.list.getClass());
+                AbstractReplicaCollection.ReplicaList subList = this.test.list.subList(from, to);
+                if (!(test instanceof ReplicaCollection.Builder<?>))
+                    Assert.assertSame(subList.contents, subCollection.list.contents);
                 Assert.assertEquals(subList, subCollection.list);
             }
         }
 
+        private void assertSubSequence(Iterable<Replica> subSequence, int from, int to)
+        {
+            AbstractReplicaCollection.ReplicaList subList = this.test.list.subList(from, to);
+            if (!elementsEqual(subList, subSequence))
+            {
+                elementsEqual(subList, subSequence);
```

Comment:
Did you mean something like `Assert.assertSame(subList.contents, subSequence);` ?

Reply:
This should simply not be there at all - the assertion below is all that's meant to be in this test.  Thanks!